common/Unit.cpp

```diff
@@ -75,31 +96,43 @@ UnitBase** UnitBase::linkAndCreateUnit(UnitType type, const std::string& unitLib
             break;
     }
 
-    if (!createHooks)
-    {
-        multiTest = false;
-        createHooks = reinterpret_cast<CreateUnitHooksFunction*>(dlsym(DlHandle, symbol));
-    }
+    CreateUnitHooksFunction* createHooks =
+        reinterpret_cast<CreateUnitHooksFunction*>(dlsym(DlHandle, symbol));
 
     if (!createHooks)
     {
         LOG_ERR("No " << symbol << " symbol in " << unitLibPath);
         return nullptr;
     }
+
     UnitBase* hooks = createHooks();
-    if (multiTest)
-    {
-        return reinterpret_cast<UnitBase**>(hooks);
-    }
-    else
-    {
+    if (hooks)
         return new UnitBase* [2] { hooks, nullptr };
-    }
+
+    LOG_ERR("No wsd unit-tests found in " << unitLibPath);
 #endif
 
     return nullptr;
 }
 
+void UnitBase::filter()
+{
+    // For now, support only filtering.
+    static const char* TestOptions = getenv("COOL_TEST_OPTIONS");
+    if (TestOptions == nullptr)
+        return;
+
+    const std::string filter = Util::toLower(TestOptions);
+    for (; GlobalArray[GlobalIndex] != nullptr; ++GlobalIndex)
+    {
+        const std::string& name = GlobalArray[GlobalIndex]->getTestname();
+        if (strstr(Util::toLower(name).c_str(), filter.c_str()))
```

Comment:
Would it make sense to have a way to support exact matches here? test/test.cpp:46 is the old code, that one works with regular expressions.

Reply:
I certainly should improve this. It was a quick way to add support for filters and other options, and to improve later. 

Yes, agreed that supporting regex is a good idea, but for the simplest case, partial match is better because it's more user friendly (there is no harm in over-matching, but annoying the user to type the exact name of a test is a bit silly, IMNSHO).